mindmeld/components/entity_resolver.py

```diff
@@ -51,51 +65,189 @@ class EntityResolver:
     (usually linked to specific entries in a knowledge base).
     """
 
-    # prefix for Elasticsearch indices used to store synonyms for entity resolution
-    ES_SYNONYM_INDEX_PREFIX = "synonym"
-    """The prefix of the ES index."""
-
-    def __init__(
-        self, app_path, resource_loader, entity_type, es_host=None, es_client=None
-    ):
-        """Initializes an entity resolver
+    @classmethod
+    def validate_resolver_name(cls, name):
+        if name not in ENTITY_RESOLVER_MODEL_TYPES:
+            msg = "Expected 'model_type' in ENTITY_RESOLVER_CONFIG among {!r}"
+            raise Exception(msg.format(ENTITY_RESOLVER_MODEL_TYPES))
+        if name == "sbert_cosine_similarity" and not sbert_available:
+            raise ImportError(
+                "Must install the extra [bert] to use the built in embbedder for entity "
+                "resolution. See https://www.mindmeld.com/docs/userguide/getting_started.html")
+
+    def __new__(cls, app_path, resource_loader, entity_type, **kwargs):
```

Comment:
Why cant we use the traditional `def __init__` approach here?

Reply:
In order to not make changes in other scripts in nlp components, an EntityResolver() class is kept. However, we want to return a different class object based on config and because we cannot return from a traditional `__init__` approach, I used `__new__` which will be called even before `__init__`.

Reply:
Why dont we use the factory pattern instead for constucting the different inherited EntityResolver objects? Happy to discuss this offline further.